rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/CreateQueueDirectives.scala

```diff
@@ -80,15 +82,19 @@ trait CreateQueueDirectives {
             await(lookupOrCreateQueue(newQueueData))
 
             queueURL(queueName) { url =>
-              respondWith {
-                <CreateQueueResponse>
-                  <CreateQueueResult>
-                    <QueueUrl>{url}</QueueUrl>
-                  </CreateQueueResult>
-                  <ResponseMetadata>
-                    <RequestId>{EmptyRequestId}</RequestId>
-                  </ResponseMetadata>
-                </CreateQueueResponse>
+              protocol match {
```

Comment:
that's a bigger refactoring but maybe ... we could also have a serializer for `CreateQueueResponse` into XML, and then we could keep the logic of creating the `CreateQueueResponse` instance here, and just call a function with the response + `protocol` + the two implicit serializers (for json and xml)

The code could be a bit cleaner I guess ... but maybe the implicit xml serializers are too much overhead

Reply:
we plan to do that, but we want to make it work first

Reply:
Right, that's probably a good idea :)